app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,436 @@
+        container: {
+            flex: 1,
+        },
+        searchBar: {
+            marginHorizontal: 12,
+            borderRadius: 8,
+            marginTop: 12,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        },
+        searchBarInput: {
+            color: theme.centerChannelColor,
+        },
+        loadingContainer: {
+            flex: 1,
+            justifyContent: 'center' as const,
+            alignItems: 'center' as const,
+        },
+        loading: {
+            height: 32,
+            width: 32,
+            justifyContent: 'center' as const,
+        },
+    };
+});
+
+export default function BrowseChannels({
+    componentId,
+    canCreateChannels,
+    joinedChannels = [],
+    sharedChannelsEnabled,
+    closeButton,
+    currentUserId,
+    currentTeamId,
+    canShowArchivedChannels,
+    categoryId,
+}: Props) {
+    const intl = useIntl();
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+
+    const [channels, setChannels] = useState<Channel[]>([]);
+    const [archivedChannels, setArchivedChannels] = useState<Channel[]>([]);
+    const [sharedChannels, setSharedChannels] = useState<Channel[]>([]);
+
+    const [visibleChannels, setVisibleChannels] = useState<Channel[]>([]);
+    const [typeOfChannels, setTypeOfChannels] = useState(PUBLIC);
+
+    const [term, setTerm] = useState('');
+
+    const [adding, setAdding] = useState(false);
+    const [loading, setLoading] = useState(false);
+
+    const mounted = useRef(false);
+    const publicPage = useRef(-1);
+    const sharedPage = useRef(-1);
+    const archivedPage = useRef(-1);
+    const nextPublic = useRef(true);
+    const nextShared = useRef(true);
+    const nextArchived = useRef(true);
+
+    const setHeaderButtons = useCallback((createEnabled: boolean) => {
+        const buttons = {
+            leftButtons: [makeLeftButton(closeButton)],
+            rightButtons: [] as OptionsTopBarButton[],
+        };
+
+        if (canCreateChannels) {
+            buttons.rightButtons = [{...makeRightButton(theme, intl.formatMessage), enabled: createEnabled}];
+        }
+
+        setButtons(componentId, buttons);
+    }, [closeButton, canCreateChannels, intl.locale, theme, componentId]);
+
+    const onSelectChannel = useCallback(async (channel: Channel) => {
+        setHeaderButtons(false);
+        setAdding(true);
+
+        const result = await joinChannel(serverUrl, currentUserId, currentTeamId, channel.id, '', false);
+
+        if (result.error) {
+            alertErrorWithFallback(
+                intl,
+                result.error,
+                {
+                    id: 'mobile.join_channel.error',
+                    defaultMessage: "We couldn't join the channel {displayName}. Please check your connection and try again.",
+                },
+                {
+                    displayName: channel ? channel.display_name : '',
```

Comment:
```suggestion
                    displayName: channel.display_name,
```

I don't think channel can be either null or undefined here